pkg/logger/logger.go

```diff
@@ -15,37 +15,58 @@ type Log struct {
 	logError *log.Logger
 	logInfo  *log.Logger
 	logDebug *log.Logger
+	logTrace *log.Logger
 }
 
 // New Set Default log level to info in case log level to defined.
 func New(logLevel string, logFilePath string) *Log {
+	// Initialize loggers with discard output
 	logError := log.New(io.Discard, "ERROR: CrowdsecBouncerTraefikPlugin: ", log.Ldate|log.Ltime)
 	logInfo := log.New(io.Discard, "INFO: CrowdsecBouncerTraefikPlugin: ", log.Ldate|log.Ltime)
 	logDebug := log.New(io.Discard, "DEBUG: CrowdsecBouncerTraefikPlugin: ", log.Ldate|log.Ltime)
+	logTrace := log.New(io.Discard, "TRACE: CrowdsecBouncerTraefikPlugin: ", log.Ldate|log.Ltime)
 
-	logError.SetOutput(os.Stderr)
-	logInfo.SetOutput(os.Stdout)
 	// we initialize logger to STDOUT/STDERR first so if the file logger cannot be initialized we can inform the user
-	if logLevel == "DEBUG" {
-		logDebug.SetOutput(os.Stdout)
-	}
+	output := os.Stdout
+	errorOutput := os.Stderr
+
+	// prepare file logging if specified
 	if logFilePath != "" {
 		logFile, err := os.OpenFile(filepath.Clean(logFilePath), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
-		if err != nil {
-			_ = fmt.Errorf("LogFilePath is not writable %w", err)
+		if err == nil {
+			output = logFile
+			errorOutput = logFile
 		} else {
-			logInfo.SetOutput(logFile)
-			logError.SetOutput(logFile)
-			if logLevel == "DEBUG" {
-				logDebug.SetOutput(logFile)
-			}
+			_ = fmt.Errorf("LogFilePath is not writable %w", err)
 		}
 	}
 
+	// Set error logger output
+	logError.SetOutput(errorOutput)
+
+	// Configure log levels
+	switch logLevel {
+	case "ERROR":
+		// Only error logging is enabled
+	case "INFO":
+		logInfo.SetOutput(output)
+	case "DEBUG":
+		logInfo.SetOutput(output)
+		logDebug.SetOutput(output)
+	case "TRACE":
+		logInfo.SetOutput(output)
+		logDebug.SetOutput(output)
+		logTrace.SetOutput(output)
```

Comment:
About Trace, We always asks in Issue debug logs, it's better for us that we have everything logged for throubleshoot with One debug level